### PR TITLE
linkFarm: add entries to passthru

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Description of changes

This is pretty handy, since it lets you reference the linked drvs through the
farm.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/build-support/trivial-builders.nix [hunk 1]
```diff
@@ -472,14 +472,18 @@ rec {
...
-      cd $out
-      ${lib.concatMapStrings (x: ''
-          mkdir -p "$(dirname ${lib.escapeShellArg x.name})"
-          ln -s ${lib.escapeShellArg x.path} ${lib.escapeShellArg x.name}
-      '') entries}
-    '';
+   linkFarm = name: entries: runCommand name {
+     preferLocalBuild = true;
+     allowSubstitutes = false;
+     passthru = lib.listToAttrs (map (x: lib.nameValuePair x.name x.path) entries);
```
[the remark this answers is not in the document]
Comment (lovesegfault) [reply pkgs/build-support/trivial-builders.nix]: Done both :)

File: pkgs/build-support/trivial-builders.nix [hunk 2]
```diff
@@ -472,14 +475,27 @@ rec {
...
-      cd $out
-      ${lib.concatMapStrings (x: ''
-          mkdir -p "$(dirname ${lib.escapeShellArg x.name})"
-          ln -s ${lib.escapeShellArg x.path} ${lib.escapeShellArg x.name}
-      '') entries}
-    '';
+  linkFarm = name: entries:
+  let
+    entries' =
+      if (lib.isAttr entries) then entries
```
Comment (roberth) [inline pkgs/build-support/trivial-builders.nix:481]:
```suggestion
      if (lib.isAttrs entries) then entries
```

**Review by roberth — COMMENTED** [review]
Clearly you're missing out on all the fun! Check out `pkgs/test/default.nix`; we're writing an awesome test suite.

**Review by roberth — APPROVED** [review]

**Comment by roberth** [comment]
Thanks!

**Pull request merged**